core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -97,18 +101,24 @@ public void handleResolvedAddressGroups(
     Map<String, Object> serviceConfig =
         attributes.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
     if (serviceConfig != null) {
-      String stickinessMetadataKey =
-          ServiceConfigUtil.getStickinessMetadataKeyFromServiceConfig(serviceConfig);
-      if (stickinessMetadataKey != null) {
-        if (stickinessMetadataKey.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
-          helper.getChannelLogger().log(
-              ChannelLogLevel.WARNING,
-              "Binary stickiness header is not supported. The header \"{0}\" will be ignored",
-              stickinessMetadataKey);
-        } else if (stickinessState == null
-            || !stickinessState.key.name().equals(stickinessMetadataKey)) {
-          stickinessState = new StickinessState(stickinessMetadataKey);
+      try {
+        String stickinessMetadataKey =
+            ServiceConfigUtil.getStickinessMetadataKeyFromServiceConfig(serviceConfig);
+        if (stickinessMetadataKey != null) {
+          if (stickinessMetadataKey.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
+            helper.getChannelLogger().log(
+                ChannelLogLevel.WARNING,
+                "Binary stickiness header is not supported. The header \"{0}\" will be ignored",
+                stickinessMetadataKey);
+          } else if (stickinessState == null
+              || !stickinessState.key.name().equals(stickinessMetadataKey)) {
+            stickinessState = new StickinessState(stickinessMetadataKey);
+          }
         }
+      } catch (Exception e) {
```

Comment:
I dont think you can catch this here.  Even if you *eventually* plan on removing this, that doesn't mean you can break it now.  This should rethrow as a RuntimeException.